### PR TITLE
doc: add release schedule and support policy

### LINKED ISSUE #341: Add a new page for release schedule and support policy for Notary Project tooling

### PR DESCRIPTION
**Pull request by yizha1**
Resolve #341 

Signed-off-by: Yi Zha <yizha1@microsoft.com>

### CODE REVIEW

**Comment by yizha1** [comment]
@sajayantony  Please take a look.

File: content/en/docs/release.md
Comment (sajayantony) [inline content/en/docs/release.md:16]: Release has a cost associated with it and maybe 4 months is more reasonable than 3. But that's something that @notaryproject/notaryproject-org-maintainers  should weigh in on.

File: content/en/docs/release.md
Comment (sajayantony) [inline content/en/docs/release.md:32]: @priteshbandi @shizhMSFT @toddysm  are you folks ok with this?

File: content/en/docs/release.md
Comment (FeynmanZhou) [reply content/en/docs/release.md:16]: Quarter-based release cadence sounds reasonable to me since the Notary Project is still in a fast iteration period. The Kubernetes community kept a quarter-based cadence for several years and changed the release cadence from four releases a year (once a quarter) to three releases a year. Maybe we can consider a 4-month release cycle after v2.

File: content/en/docs/release.md
Comment (shizhMSFT) [inline content/en/docs/release.md:39]: This row confuses me. Can you elaborate more?

Comment (yizha1) [reply content/en/docs/release.md]: @shizhMSFT Would you mind reviewing it again to see whether it gets clarified.

**Comment by yizha1** [comment]
@priteshbandi @iamsamirzon would you mind reviewing this PR?

File: content/en/docs/release.md
Comment (shizhMSFT) [reply content/en/docs/release.md:14]: The reason is that I'm confused with the TOC.

Should it be something like

- Release versioning
  - Major releases
  - Minor releases
  - Patch releases
- Release notes
- Support policy
- Upcoming releases
- Attribution

File: content/en/docs/release.md
Comment (shizhMSFT) [inline content/en/docs/release.md:39]: just a note: patch releases can be after `v1.1.0`.

File: content/en/docs/release.md
Comment (yizha1) [reply content/en/docs/release.md:16]: Considering we still have many backlog issues not well addressed, so I would suggest quarterly based releases.

**Comment by yizha1** [comment]
@priteshbandi @iamsamirzon @sajayantony @shizhMSFT @FeynmanZhou  Would you mind reviewing this PR again?

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by FeynmanZhou — APPROVED** [review]
LGTM

File: content/en/docs/release.md
Comment (FeynmanZhou) [inline content/en/docs/release.md:8]:
```suggestion
Since the Notary Project v1.0.0 release, subsequent Notary Project releases are scheduled quarterly.
```

Is it more accurate?

File: content/en/docs/release.md
Comment (iamsamirzon) [inline content/en/docs/release.md:16]:
```suggestion
Patch releases provide users with bug fixes and security fixes. They do not contain new features. A patch release can be done by each Notary Project repository individually every quarter in a calendar year. A patch release to fix critical issues or security issues can be done on demand. ~~A patch release can be cancelled if there is no new fixes since the previous release.~~
```
The last line is not needed. It is more confusing to say "Patch release can be cancelled" then not to say anything. It is obvious that if no fixes are there, no patch release will be created. The cancelling word may imply that we may do a patch release but can redact it later.

File: content/en/docs/release.md
Comment (iamsamirzon) [inline content/en/docs/release.md:20]:
```suggestion
Minor releases contain new features, security and bug fixes. They are backwards compatible with respect to the API and the CLI usage. A minor release may be done every quarter in a calendar year. If a feature needs specifications changes, a corresponding release for specifications will be planned. If a feature needs documentation, documentation need to be published on [Notary Project website](https://notaryproject.dev/docs/) during a minor release. Code freeze starts one week before a minor release. Only critical bug fixes are accepted into the release codebase during this time. Features that are planned but incomplete before the the minor release will be moved to next minor release.
```

**Pull request closed without merging**
